File: backend/app/schemas/validation.py
from pydantic import BaseModel
from typing import Literal


class ValidationComment(BaseModel):
    id: str
    author: str
    text: str
    createdAt: str


class ValidationItem(BaseModel):
    id: str
    documentId: str
    documentName: str
    confidence: float
    conflictReason: str
    reviewer: str | None = None
    status: Literal["pending", "approved", "rejected"]
    comments: list[ValidationComment]
    createdAt: str

File: frontend/src/types/validation.ts
export type ValidationItemStatus =
  | 'pending'
  | 'approved'
  | 'rejected'
  | 'pending_review'
  | 'under_review'
  | 'human_verified'
  | 'auto_verified';

export interface ValidationComment {
  id: string;
  author: string;
  text: string;
  createdAt: string;
}

export interface ValidationItem {
  id: string;
  documentId: string;
  documentName: string;
  confidence: number; // 0-100
  conflictReason: string;
  reviewer?: string;
  status: ValidationItemStatus;
  comments: ValidationComment[];
  createdAt: string;
}
